### PR TITLE
CSS-3455: Add safe upgrades integration test

### PR DESCRIPTION
**Pull request by kelkawi-a**
This PR adds another integration test to check for the ability to safely upgrade to the newly released charm. The integration test deploys the charm from the latest channel available on Charmhub and upgrades it to the charm built from source. This ensures that the changes made in each PR are safe to upgrade to from the previous release.

### CODE REVIEW

**Review by gtato — APPROVED** [review]
lgtm

File: tests/integration/test_upgrades.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        """Builds the current charm and refreshes the current deployment."""
+        charm = await ops_test.build_charm(".")
+        resources = {"temporal-server-image": METADATA["containers"]["temporal"]["upstream-source"]}
+
+        await ops_test.model.wait_for_idle(apps=[APP_NAME], status="active", raise_on_blocked=False, timeout=600)
+
+        await ops_test.model.applications[APP_NAME].refresh(path=str(charm), resources=resources)
+
+        async with ops_test.fast_forward():
+            time.sleep(10)  # time for application to settle
```
Comment (kian99) [inline tests/integration/test_upgrades.py:71]: Do you need this sleep if you have the `wait_for_idle` below?

File: tests/integration/test_upgrades.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            await ops_test.model.wait_for_idle(apps=[APP_NAME], status="active", raise_on_blocked=False, timeout=600)
+            await ops_test.model.wait_for_idle(status="active")
```
Comment (kian99) [inline tests/integration/test_upgrades.py:73]: Also does `wait_for_idle` need to be repeated?

**Review by kian99 — APPROVED** [review]

File: tests/integration/test_upgrades.py (hunk 1 above)
Comment (kelkawi-a) [reply tests/integration/test_upgrades.py]: I added a comment in the latest commit, but the unit was becoming active while the application remained in a waiting state for a few seconds. This led to the test failing when the sample workflow is run, therefore the delay is added to allow the application to go into an active state as well before running the workflow.

**Pull request merged**